cpp/include/raft/neighbors/sample_filter.cuh

```diff
@@ -65,7 +65,7 @@ struct NoneSampleFilter {
  * };
  *
  * Initialize it as:
- *   using filter_type = IndexSampleFilter<idx_t>;
```

Comment:
Instead of `raft/neighbors/sample_filter.cuh`, we use the suffix `_types` to denote header files that contain important vocabulary bits for interacting w/ any public APIs not in `raft/core`. Can you rename this file to `neighbors/sample_filter_types.hpp`? It's also important that any functions in these files be accessible from host (and conditionally accessible from device). Can you change all `__host__ __device__` to `_RAFT_HOST_DEVICE` (from `core/detail/macros.hpp`)?

Reply:
@cjnolet sure, 23.08.